test/index.js

```diff
@@ -271,6 +271,16 @@ describe('maybe', function() {
                             'Function as its first argument; received null'));
     });
 
+    it('provides a "reduce" method', function() {
+      eq(S.Nothing().reduce.length, 2);
+      eq(S.Nothing().reduce(function(a, b) { return a + b; }, 10), 10);
+
+      assert.throws(function() { S.Nothing().reduce(null); },
+        errorEq(TypeError,
+          '‘Nothing#reduce’ requires a value of type ' +
+          'Function as its first argument; received null'));
```

Comment:
I find this nesting makes it easier to see what's being applied to what, since the arguments are aligned:

```javascript
assert.throws(function() { S.Nothing().reduce(null); },
              errorEq(TypeError,
                      '‘Nothing#reduce’ requires a value of type ' +
                      'Function as its first argument; received null'));
```

I've been referring to this as Lisp-style indentation, though I'm not sure this is accurate. :)